dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -271,59 +293,78 @@ internal constructor(
     private fun updateRequest(
         request: Request,
         tracer: Tracer,
-        span: Span
+        span: Span?
     ): Request.Builder {
         val tracedRequestBuilder = request.newBuilder()
 
-        tracer.inject(
-            span.context(),
-            Format.Builtin.TEXT_MAP_INJECT,
-            TextMapInject { key, value ->
-                // By default the `addHeader` method adds a value and doesn't replace it
-                // We need to remove the old trace/span info to use the one for the current span
-                tracedRequestBuilder.removeHeader(key)
-                tracedRequestBuilder.addHeader(key, value)
+        if (span == null) {
+            listOf(
+                SAMPLING_PRIORITY_HEADER,
+                TRACE_ID_HEADER,
+                SPAN_ID_HEADER
+            ).forEach {
+                tracedRequestBuilder.removeHeader(it)
             }
-        )
+            tracedRequestBuilder.addHeader(SAMPLING_PRIORITY_HEADER, DROP_SAMPLING_DECISION)
+        } else {
+            tracer.inject(
```

Comment:
shouldn't you add the `SAMPLING_PRIORITY_HEADER` here with `1` ? Otherwise when you check for it in the `TracingInterceptor` the way I've seen you are going to apply a sampling there which is not wanted if the request was not sampled out here.

Reply:
Good question :) I was thinking about it, but let it as it is now. So normally the expectation is that `Injector` will add it, because this is what `DatadogHttpCodec` is doing here https://github.com/DataDog/dd-sdk-android/blob/c05f878ec4d84c65d70aedacb20c6bce6cc9c9ba/dd-sdk-android/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java#L54.

We could check if header is there after the injection and if it is not, we could add it, but the thing is that `Request.Builder` has no method to check if header is already there, and calling `removeHeader` + `addHeader` (because simple `addHeader` just adds it and not replacing if there is an existing one) can set a header with a value different from injector. But anyway, with `DatadogHttpCodec` everything should be fine.

Now the question is: what if customer is using non-Datadog tracer (generally speaking it is possible, because we accept high-level `io.opentracing.Tracer` interface, not necessary Datadog tracer. So it won't be `DatadogHttpCodec` there and so maybe `x-datadog-sampling-priority` won't be set in this branch. If it is not set, then simply on the backend side service will make its own decision about the sampling, so not a big deal.

And since we are accepting `io.opentracing.Tracer` and `io.opentracing.Span` interface which generally **have** a way to set priority sampling, but there is no way to read it, we have to rely on the headers with such kind of hacky decision.

We could assume that Datadog tracer is always used (and then use `DDSpan`, which have an opportunity to read back the sampling priority) and not rely on the headers, but then `traceSamplingRate` parameter would be useless if customer is using non-Datadog tracer and so they will face the new billing model in a not very nice way.

Reply:
This is done automatically by the inject method